arch/x86/include/intel64/kernel_arch_thread.h

```diff
@@ -34,6 +36,8 @@ struct _callee_saved {
 typedef struct _callee_saved _callee_saved_t;
 
 struct _thread_arch {
+	u8_t flags;
+
```

Comment:
Given there are only two flags in use and everything else in this struct is 8-byte aligned, you could save a few cycles by making them into byte values that can be accessed with a single load.